nats.go

```diff
@@ -3268,17 +3275,58 @@ func badQueue(qname string) bool {
 }
 
 // subscribe is the internal subscribe function that indicates interest in a subject.
-func (nc *Conn) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync bool) (*Subscription, error) {
+func (nc *Conn) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync bool, opts ...SubscribeOption) (*Subscription, error) {
 	if nc == nil {
 		return nil, ErrInvalidConnection
 	}
+
+	var aopts *jsOpts
+	if len(opts) > 0 {
```

Comment:
Should I fail here if old style subscriptions are not used?

Reply:
Not following, fail why?

Reply:
Because as discussed, much of JS will not work with it.

Reply:
This is just low level subscribe. We aren't doing the mux inbox trick here, just on Request().

Reply:
JS works with new style unless multiple concurrent requests in flight.

Reply:
If a user specify a NewInbox() on a consumer as in the tests, we really cant says "unless multiple concurrent" we have no way of knowing when that is happening.  The consequence is literally messages go missing and never get delivered, we cant have that.

Reply:
`NewInbox()` just creates a subject that you still need to subscribe to IIRC. So I think we are fine there.